txserver/quote.go

```diff
@@ -31,14 +31,14 @@ func get_quote(username string, stock string) string {
 		conn = quote_server_connect()
 	}
 
-	_, err := conn.Write([]byte(stock + username))
+	_, err := conn.Write([]byte(stock + username + "\n"))
 	if err != nil {
-		return get_quote(username, stock)
+		return get_quote(stock, username)
 	}
 
 	result, err := ioutil.ReadAll(conn)
 	if err != nil || result == nil {
-		return get_quote(username, stock)
+		return get_quote(stock, username)
```

Comment:
there is a possibility of infinite recursion here. We should retry for a certain number of times.